automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/WorkspaceMonitorPage.scala

```diff
@@ -27,7 +27,11 @@ class WorkspaceMonitorPage(namespace: String, name: String)(implicit webDriver:
   }
 
   def openSubmission(submissionId: String): Unit = {
-    submissionLink(submissionId).doClick()
-    await ready new SubmissionDetailsPage(namespace, name, submissionId)
+    val url = webDriver.getCurrentUrl
+    // don't click if already on Submission Detail page
+    val page = new SubmissionDetailsPage(namespace, name, submissionId)
+    if (url != page.url)
+      submissionLink(submissionId).doClick()
+    await ready page
```

Comment:
Seems like a logic error in a test. Which test is causing the problem?

Reply:
can't remember which test now. but I can revert if you feel very strongly against change.

Reply:
Yeah, this is a really bad code smell to me.